### ISSUE TITLE
Incorrect formatting in output when run using '!' inside nvim [Not in vim].

### ISSUE THREAD

**Issue by pallavagarwal07** [neovim/neovim#2958]
If I do 

```
:!echo -e "1\t2\t3"
```

output shown by vim is:

```
0   1   2
```

While that by nvim is: 

```
0 1 2
```

Due to this, many scripts that rely on tabs for basic formatting are being displayed incorrectly.

EDIT: I just found that echo works this way only in zsh. To reproduce the error on bash have a python script `print "1\t2\t3"` and run it from inside `nvim` using `:!python %`

I'm using:
NVIM 0.0.0-alpha+201507051031
Terminal: Urxvt v9.21

**Comment by atommclain** [comment]
9a3b7f Gentoo 2.2 Bash 4.3.42 connected via iTerm 2.1.1 

`parse_control_character` in `ui.c` does not handle tabs.

I believe the only reason there is a space is because of the way `send_output` parses double cell characters. This may be related to the issue https://github.com/neovim/neovim/issues/4338.

While researching this ticket I discovered the *nix command `tabs` which allows a user to set their own custom tab settings. ex `tabs 5 16`. I used those tab settings then followed the original repro settings with vim 7.4.769 and received the following output:
```
1     2                    3 // Graphically selecting whitespace revealed one tab character of different length
                             // where it is faked here with multiple spaces
```
This will be important incase a control character function like `ui_horitzontal_tab` is added to `ui.c` as tabs at different levels of indentation may be represented by different lengths.

Resources:
The *nix command `infocmp` shows the tab settings
http://stackoverflow.com/questions/10782699/how-to-set-4-space-tab-in-bash
